bolt-simple-taproot.md

```diff
@@ -0,0 +1,994 @@
+the spending conditions do not require a script path, the output key should
+commit to an unspendable script path instead of having no script path. This can
+be achieved by computing the output key point as `Q = P +
+int(hashTapTweak(bytes(P)))G`". We refer to BIP 86 for the specifics of this
+derivation.
+
+By using the musig2 sorting routine, we inherit the lexicographical key
+ordering from the base segwit channels. Throughout the lifetime of a channel
+any keys aggregated via musig2 also are assumed to use the `KeySort` routine
+
+### Commitment Transactions
+
+#### To Local Outputs
+
+For the simple taproot commitment type, we use the same flow of revocation or
+delay, but instead left the revocation case into the taproot key-spend case.
+
+The new output has the following form:
+
+  * `OP_1 to_local_output_key`
+  * where:
+    * `to_local_output_key = revocationpubkey + tagged_hash("TapTweak", revocationpubkey || to_delay_script_root)`
+    * `to_delay_script_root = tapscript_root([to_delay_script])`
+    * `to_delay_script` is the delay script:
+        ```
+        <local_delayedpubkey> OP_CHECKSIG
+        <to_self_delay> OP_CHECKSEQUENCEVERIFY OP_DROP
+        ```
+
+The parity (even or odd) of the y-coordinate of the derived
+`to_local_output_key` MUST be known in order to derive a valid control block.
+
+The `tapscript_root` routine constructs a valid taproot commitment according to
+BIP 341+342. Namely, a leaf version of `0xc0` MUST be used. As there's only a
+single script, one can derive the tapscript root as:
+```
+tapscript_root = tagged_hash("TapLeaf", 0xc0 || compact_size_of(to_delay_script) || to_delay_script)
+```
+
+In the case of a commitment breach, the `to_delay_script_root` can be used
+along side `<revocationpubkey>` to derive the private key needed to sweep the
+top-level key spend path. A valid witness is then just:
+```
+<revocation_sig>
+```
+
+In the case of a routine force close, the script path must be revealed so the
+broadcaster can sweep their funds after a delay. The control block to spend is
+only `33` bytes, as it just includes the internal key (along with the y-parity
+bit and leaf version):
+```
+delay_control_block = (output_key_y_parity | 0xc0) || revocationpubkey
+```
+
+A valid witness is then:
+```
+<local_delayedsig> <to_delay_script> <delay_control_block>
+```
+
+As with base channels, the `nSequence` field must be set to `to_self_delay`.
+
+#### To Remote Outputs
+
+As we inherit the anchor output semantics we want to ensure that the remote
+party can unilaterally sweep their funds after the 1 block CSV delay. In order
+to achieve this property, we'll re-use the `_funding_key` here: its in
+the best interest of the other party to enforce these semantics (mempool
+pinning mitigation), so the remote party will be forced to always take the
+script reveeal path.
+
+The to remote output has the following form:
+
+  * `OP_1 to_remote_output_key`
```

Comment:
What exactly do you mean by this? A Taproot output, as far as I understand, can only have the form `0x51 <32-byte-pubkey>`. This looks like there's also an `OP_1` prefix for the `OP_CHECKSEQUENCEVERIFY` command, but it seems to exist outside the tap tree structure? Can you maybe give an example with sum pubkeys of what the Taproot Merkle tree and output script would look like?

Reply:
I dont see what is wrong with this. OP_1 == 0x51 and the 32 byte pubkey in this case is the to remote output key

Reply:
My confusion partially stems from the `to_remote_script` semantics. In BOLT 3, the script is `<remotepubkey> OP_CHECKSIGVERIFY 1 OP_CHECKSEQUENCEVERIFY`. Here, however, the input to `OP_CSV` seems to be dependent on the validity of the signature against `remotepubkey`. So if an invalid signature is presented, the script stack would end up as `0 OP_CSV`?

Reply:
The script can only succeed with a valid signature, and then the sequence is checked. The OP_1 from the taproot _output_ has nothing to do with the 1 block relative timelock

Reply:
Right, disregarding the OP_1 (which is indeed simply the witness version, sorry for the distraction), my concern was that `O OP_CSV` wouldn't fail, but given that even though the `OP_CSV` part doesn't fail per se, it leaves a falsey stack element at the top, so all does check out in the end. I'll close this entire thread.